### ISSUE TITLE
source code in this repo is outdated

### ISSUE THREAD

**Issue by GitHubIsToxic** [Ericsson/ETCPACK#12]
The etcpack executable bundled with the Mali Texture Compression package reports:
`ETCPACK v4.0.1 for ETC and ETC2`

And the code in this GitHub repo reports:
`ETCPACK v2.74 For ETC and ETC2`

I cannot find the source code of the newer version anywhere online. The new version is claimed to be much faster per Mali Texture Compression webpage:

> Improved ETC1 compression speed of up to 600x compared to ETCPack at the same or better quality.

(https://developer.arm.com/tools-and-software/graphics-and-gaming/mali-texture-compression-tool)

Has etcpack become closed source?

**Comment by jacobstrom-ericsson** [comment]
Hi!

Sorry for the extremely late reply. I was searching my inbox for unread messages and found this one.

I am not personally aware of any version newer than v2.74. It is possible that someone working on the Mali package has forked ETCPACK and created newer versions. That is something that I have no knowledge of however.

I will try to be more responsive in the future, apologies again for the very late reply.

Best Regards,

-Jacob

**Comment by GitHubIsToxic** [comment]
Thanks Jacob, no problem. I will ask someone from ARM if they are willing to release the code.

FYI, I ended up using rg_etc1 by Rich Geldreich (https://github.com/richgel999/rg-etc1), which was trivial to make multi-threaded, and results in high quality compression, far better than etcpak 0.6 (https://github.com/thedmd/etcpak), but also slower. I haven't compared it to ETCPACK.

**Comment by QinGeneral** [comment]
@GitHubIsToxic Have you got the code of Mali Texture Compression's etcpack? I want it too......Thanks.

**Comment by jacobstrom-ericsson** [comment]
Hi 

@GitHubIsToxic, if you are after maximum quality, then ETCPACK has an exhaustive mode which guarantees optimum quality under PSNR distortion metric. However, that is seldom what people are after since it is very slow! :-)

I will now close this question.

**Comment by GitHubIsToxic** [comment]
@QinGeneral: no, I have only what's in this very Github repo.

@jacobstrom-ericsson: I found ETCPACK very hard to integrate in my application (never properly got it working in my code base), so I ended up with rg_etc1 with added multi-threading. It's still sub-optimal as it never goes above ~66% on all 12 cores on my machine (highest non-exhaustive mode), but it's much faster than plain rg_etc1. I didn't do any accurate benchmarking though, I just decided it was 'fast enough' for my purposes. It's possible ETCPACK is still faster at similar quality level.
